### ISSUE TITLE
How to enable Textobject queries for TS and JS?

### ISSUE THREAD

**Issue by moonmoonCL** [helix-editor/helix#3177]
How to enable Textobject queries for TS and JS?
<img width="814" alt="image" src="https://user-images.githubusercontent.com/55181046/180646528-4c89de0e-9d4e-4452-8a67-e8b037e5d3ad.png">

**Comment by A-Walrus** [comment]
Textobject queries are not something that can be enabled by the user, rather someone has to write and add a textobject query file for JS/TS to Helix at `runtime/queries/javascript/textobjects.scm` and `runtime/queries/typescript/textobjects.scm` with the necessary queries

**Comment by moonmoonCL** [comment]
Thanks, where could I find that file?

**Comment by A-Walrus** [comment]
The file doesn't exist yet, someone needs to write it :) Until someone does so, textobject queries for JS/TS will not be available on Helix.

**Comment by the-mikedavis** [comment]
Let's keep this open since these may be contributed by someone looking through the issue list 🙂 

https://docs.helix-editor.com/master/guides/textobject.html

**Comment by adamkpickering** [comment]
I cannot get `mic` to work with the following typescript class. I've tried a number of things, but cannot figure out anything that will make it work... is there any chance we can reopen this? Or maybe it should be a new issue.
```
import fs from 'fs';

export default class DependencyVersions {
  alpineLimaISO: string = "";
  WSLDistro: string = "";
  kuberlr: string = "";
  helm: string = "";
  docker: string = "";
  dockerBuildx: string = "";
  dockerCompose: string = "";
  trivy: string = "";
  steve: string = "";
  dockerProvidedCredentialHelpers: string = "";
  ECRCredenialHelper: string = "";
  hostResolver: string = "";
  
  constructor(inputObject: any) {
    for (const key in this) {
      const inputValue = inputObject[key];
      if (!inputValue) {
        throw new Error(`key "${ key }" from input object is falsy`);
      }
      Reflect.set(this, key, inputValue);
    }
  }

  static async fromJSONFile(path: string) {
    const rawContents = await fs.promises.readFile(path, 'utf8')
    const obj = JSON.parse(rawContents);
    return new DependencyVersions(obj);
  }
}
```

**Comment by the-mikedavis** [comment]
The new textobjects work fine for me https://asciinema.org/a/512473

Your queries may be out of date. Are you building from the latest source? Also make sure your [runtime directory has been symlinked](https://github.com/helix-editor/helix/blob/c5f8a835f54b27c9f1c22c8321378a309d371309/README.md?plain=1#L48-L55) from the latest source (rather than copying which will let the queries be stale).

**Comment by adamkpickering** [comment]
Yes, I had built from the latest source, and symlinked the runtime directory. But funny enough, it is working for me today. At some point yesterday, `hx` locked up and I had to restart - I'm thinking maybe something behind the scenes got into a bad state that was causing `mic` not to work, and restarting `hx` fixed that. :shrug:
